### DISCUSSION TITLE
Ansteuerung der Wendefunktion über vcaSetState

### DISCUSSION THREAD

**Discussion by jason0611** [Mogli12/KeyboardSteer#760] [Ideas]
Hallo Stefan,
zunächst einmal vielen Dank für diesen großartigen Mod und die superschnelle Migration in den LS22.
Als Programmierer des Vorgewende Managements würde ich die neu hinzugekommene Wendefunktion (SNAPLEFT/SNAPRIGHT) gerne über das Vorgewende Management ansprechen, so dass, statt die GPS-Führung zu pausieren, die Wendefunktion angestoßen wird. In einem ersten POC funktioniert das sehr gut, allerdings müsste ich dazu zu viel Code direkt übernehmen, was nicht meinem Verständnis von sauberem Modding entspricht. Mir geht es  schließlich darum, Funktionen zu nutzen und nicht Code zu "übernehmen".
Ist es möglich, die Wendefunktion über den Aufruf der von Dir bereitgestellten Funktion vcaGetState zu realisieren? Dann könnte ich einfach diese Schnittstelle ansprechen und die Umsetzung wäre an dieser Stelle sauber.
Viele Grüße,
Martin

**Comment by Mogli12** [comment]
Hallo Martin,
über vcaSetState kann man das leider nicht einstellen. Aber ich kann das gleich machen. Willst Du nur die nach rechts/links Funktion aufrufen? Oder möchtest Du noch die Anzahl Spuren mit geben?
Viele Grüße,
Stefan

**Comment by jason0611** [comment]
Hallo Stefan,
links/rechts reicht aus. Die Anzahl der Spuren können ja direkt im VCA eingestellt werden.
Viele Grüße,
Martin

**Comment by Mogli12** [comment]
Hallo Martin,
ich habe mal gerade bei FS22_HeadlandManagement vorbeigeschaut. Ich kann Dir auch extra Funktionen bauen, so dass man die Geräte vorne oder hinten heben, senken, ein- und ausschalten kann.
Viele Grüße,
Stefan

**Comment by jason0611** [comment]
Hallo Stefan,
das getrennte Ausheben/Absenken und An-/Abschalten ist schon vorhanden und funktioniert recht zuverlässig. Das lässt sich derzeit für alle Geräte oder wahlweise nur vorne oder nur hinten einstellen.
Viele Grüße,
Martin

**Comment by jason0611** [comment]
Nachtrag: Soll aber keine generelle Ablehnung sein, für Optimierungen bin ich immer zu haben. :-)

**Comment by Mogli12** [comment]
Hallo Martin,

In FS22 Build 21 sind jetzt die beiden Funktionen 
```lua
		self:vcaSnapReverseLeft()
		self:vcaSnapReverseRight()
```
enthalten. Das Heben/Senken habe ich zufällig bei Dir gesehen, daher die Idee. 

Viele Grüße,
Stefan

**Comment by jason0611** [comment]
Perfekt, Danke Dir.
Das baue ich direkt ein und kann dann den testweise übernommenen Code direkt wieder ausräumen. :-)

**Comment by Mogli12** [comment]
Es ist auf jeden Fall besser, wenn das über eine separate Funktion läuft. Das Feature werde ich sicher nicht ausbauen. Aber eventuell baue unterhalb noch einiges um.

**Comment by jason0611** [comment]
Definitiv. Daher nutze ich ja auch am liebsten vorgegebene Schnittstellen, dann ist die Funktion unabhängiger.
Ich teste das morgen nochmal gründlich und gebe dann eine kurze Rückmeldung. :-)

**Comment by jason0611** [comment]
Hallo Stefan,
wie erwartet, funktioniert der Aufruf über die beiden Funktionen tadellos und fügt sich gut ins Vorgewende Management ein. Nochmals vielen Dank für die schnelle Umsetzung.
Ich mache hier mal zu. :-)
Viele Grüße,
Martin
